### PR TITLE
Added sum with recursion to 03_recursion, java folder

### PR DESCRIPTION
**Pull request by To-heeb**
Update: Arraylist is not to be modified during recursion.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@ highlights/atom-language-perl6/
 .DS_store
 highlights/package-lock.json
 zig-cache
+assignments
```
Comment (egonSchiele) [inline .gitignore:31]: can you undo this change?

File: 03_recursion/java/04_sum/src/Sum.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import java.util.*;
+
+public class Sum {
+    public static int sum(ArrayList<Integer> num_list) {
+
+        if (num_list.size() == 0) {
+            return 0;
+        } else {
+            int num = num_list.get(0);
+            num_list.remove(0);
```
Comment (egonSchiele) [inline 03_recursion/java/04_sum/src/Sum.java:10]: could you write a version that doesn't modify the list?

**Review by egonSchiele — CHANGES_REQUESTED** [review]

File: 03_recursion/java/04_sum/src/Sum.java (hunk 2 above)
Comment (To-heeb) [reply 03_recursion/java/04_sum/src/Sum.java]: WIP

**Comment by egonSchiele** [comment]
Thanks!
